lib/recurly/apple-pay/apple-pay.js

```diff
@@ -377,3 +392,22 @@ export class ApplePay extends Emitter {
     };
   }
 }
+
+function restorePricing (pricing, state, done) {
+  if (!pricing) return done();
+
+  let reprice = false;
+
+  if (pricing.items.address != state.address) {
+    pricing.items.address = state.address;
+    reprice = true;
+  }
+
+  if (pricing.items.shippingAddress != state.shippingAddress) {
+    pricing.items.shippingAddress = state.shippingAddress;
+    reprice = true;
+  }
+
+  if (reprice) pricing.reprice(done);
+  done();
```

Comment:
`done` can be called twice here if reprice is needed. Probably just need an `else`.

Reply:
updated